iStore/Login/LoginView.swift

```diff
@@ -55,15 +47,27 @@ struct LoginView: View {
                     authenticationViewModel.showProgress = true
                     authenticationViewModel.showError = false
                     
-                    authenticationViewModel.authenticateUser(userName: userName, password: password) {
-                        navigationCoordinator.switchView = .contentView
-                        authenticationViewModel.showProgress = false
+                    authenticationViewModel.authenticateUser() { result in
+                        switch result {
+                        case .success(let currentUser):
+                            DispatchQueue.main.async {
+                                authenticationViewModel.user = currentUser
```

Comment:
Also here why do you assign the values here?

Reply:
Because It's the completion of the authencation Method.

Reply:
There is something wrong here. This logic should be handle by the model view.

Reply:
I have to do it here as it affects the UI and the navigationCoordinator is an EnvironmentObject so It's properties must be accessed from the view.